<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CompilersTechnologyClass</class>
 <widget class="QMainWindow" name="CompilersTechnologyClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>952</width>
    <height>540</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>CompilersTechnology</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QWidget" name="horizontalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>287</width>
      <height>31</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QPushButton" name="buttonPage1">
       <property name="text">
        <string>文法类型判别</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="buttonPage2">
       <property name="text">
        <string>NFA确定化与最小化</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="buttonPage3">
       <property name="text">
        <string>词法分析器</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QStackedWidget" name="stackedWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>40</y>
      <width>941</width>
      <height>461</height>
     </rect>
    </property>
    <widget class="QWidget" name="page">
     <widget class="QLabel" name="inPut">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>0</y>
        <width>381</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>11</pointsize>
       </font>
      </property>
      <property name="text">
       <string>输入</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignCenter</set>
      </property>
     </widget>
     <widget class="QPushButton" name="chargeCs">
      <property name="geometry">
       <rect>
        <x>450</x>
        <y>210</y>
        <width>80</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>判断</string>
      </property>
     </widget>
     <widget class="QTextEdit" name="csOutput">
      <property name="geometry">
       <rect>
        <x>540</x>
        <y>40</y>
        <width>381</width>
        <height>371</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="outPut">
      <property name="geometry">
       <rect>
        <x>540</x>
        <y>0</y>
        <width>381</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>11</pointsize>
       </font>
      </property>
      <property name="text">
       <string>输出</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignCenter</set>
      </property>
     </widget>
     <widget class="QPushButton" name="AddCs">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>420</y>
        <width>61</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>添加文法</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>170</x>
        <y>0</y>
        <width>41</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>15</pointsize>
       </font>
      </property>
      <property name="text">
       <string>G(E)</string>
      </property>
     </widget>
     <widget class="QPushButton" name="DeleteCs">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>420</y>
        <width>61</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>删除文法</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="VnInput">
      <property name="geometry">
       <rect>
        <x>49</x>
        <y>40</y>
        <width>160</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QLineEdit" name="VtInput">
      <property name="geometry">
       <rect>
        <x>49</x>
        <y>80</y>
        <width>160</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QLabel" name="label_5">
      <property name="geometry">
       <rect>
        <x>-1</x>
        <y>40</y>
        <width>41</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>15</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Vn</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>-1</x>
        <y>80</y>
        <width>41</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>15</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Vt</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_7">
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>40</y>
        <width>20</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>15</pointsize>
       </font>
      </property>
      <property name="text">
       <string>S</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="S">
      <property name="geometry">
       <rect>
        <x>269</x>
        <y>40</y>
        <width>160</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QScrollArea" name="WfSa">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>120</y>
        <width>381</width>
        <height>291</height>
       </rect>
      </property>
      <property name="verticalScrollBarPolicy">
       <enum>Qt::ScrollBarPolicy::ScrollBarAsNeeded</enum>
      </property>
      <property name="horizontalScrollBarPolicy">
       <enum>Qt::ScrollBarPolicy::ScrollBarAlwaysOff</enum>
      </property>
      <property name="widgetResizable">
       <bool>false</bool>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignCenter</set>
      </property>
      <widget class="QWidget" name="WifaWidget">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>401</width>
         <height>289</height>
        </rect>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <widget class="QWidget" name="gridLayoutWidget">
        <property name="geometry">
         <rect>
          <x>-2</x>
          <y>0</y>
          <width>381</width>
          <height>291</height>
         </rect>
        </property>
        <layout class="QGridLayout" name="CsInput" rowstretch="0" columnstretch="0,0,0" rowminimumheight="0" columnminimumwidth="0,0,0">
         <property name="sizeConstraint">
          <enum>QLayout::SizeConstraint::SetFixedSize</enum>
         </property>
         <property name="leftMargin">
          <number>5</number>
         </property>
         <property name="rightMargin">
          <number>5</number>
         </property>
         <property name="spacing">
          <number>10</number>
         </property>
         <item row="0" column="0">
          <widget class="QLineEdit" name="L0">
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLabel" name="CS0">
           <property name="font">
            <font>
             <pointsize>15</pointsize>
            </font>
           </property>
           <property name="text">
            <string>::=</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QLineEdit" name="R0"/>
         </item>
        </layout>
       </widget>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="page_2"/>
    <widget class="QWidget" name="page_3"/>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>952</width>
     <height>20</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="CompilersTechnology.qrc"/>
 </resources>
 <connections/>
</ui>
